drivers/ptp_clock/ptp_clock_shell.c

```diff
@@ -0,0 +1,253 @@
+
+	entry->syntax = (dev != NULL) ? dev->name : NULL;
+	entry->handler = NULL;
+	entry->help = NULL;
+	entry->subcmd = NULL;
+}
+
+SHELL_DYNAMIC_CMD_CREATE(dsub_device_name, device_name_get);
+
+static int parse_device_arg(const struct shell *sh, char **argv, const struct device **dev)
+{
+	*dev = shell_device_get_binding(argv[1]);
+	if (!*dev) {
+		shell_error(sh, "device %s not found", argv[1]);
+		return -ENODEV;
+	}
+	return 0;
+}
+
+/* ptp_clock get <device> */
+static int cmd_ptp_clock_get(const struct shell *sh, size_t argc, char **argv)
+{
+	struct net_ptp_time tm = {0};
+	const struct device *dev;
+	int ret;
+
+	ret = parse_device_arg(sh, argv, &dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = ptp_clock_get(dev, &tm);
+	if (ret < 0) {
+		return ret;
+	}
+
+	shell_print(sh, "%"PRIu64".%09u", tm.second, tm.nanosecond);
+
+	return 0;
+}
+
+/* ptp_clock set <device> <seconds> */
+static int cmd_ptp_clock_set(const struct shell *sh, size_t argc, char **argv)
+{
+	struct net_ptp_time tm = {0};
+	const struct device *dev;
+	int ret;
+
+	ret = parse_device_arg(sh, argv, &dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	tm.second = shell_strtoull(argv[2], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = ptp_clock_set(dev, &tm);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+/* ptp_clock adj <device> <seconds> */
+static int cmd_ptp_clock_adj(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int adj;
+	int ret;
+
+	ret = parse_device_arg(sh, argv, &dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	adj = shell_strtol(argv[2], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = ptp_clock_adjust(dev, adj);
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+/* ptp_clock freq <device> <ppb> */
+static int cmd_ptp_clock_freq(const struct shell *sh, size_t argc, char **argv)
+{
+	const struct device *dev;
+	int ppb;
+	int ret;
+
+	ret = parse_device_arg(sh, argv, &dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ppb = shell_strtol(argv[2], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = ptp_clock_rate_adjust(dev, 1.0 + ((double)ppb / 1000000000.0));
+	if (ret < 0) {
+		return ret;
+	}
+
+	return 0;
+}
+
+/* ptp_clock selftest <device> <time> <freq> <delay> <adj> */
+static int cmd_ptp_clock_selftest(const struct shell *sh, size_t argc, char **argv)
+{
+	struct net_ptp_time tm = {0};
+	const struct device *dev;
+	int seconds, freq, delay, adj, ret;
+
+	ret = parse_device_arg(sh, argv, &dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	seconds = shell_strtol(argv[2], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	freq = shell_strtol(argv[3], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	delay = shell_strtol(argv[4], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	adj = shell_strtol(argv[5], 10, &ret);
+	if (ret < 0) {
+		return ret;
+	}
+
+	/* set 'time' seconds and read back to verify clock setting/getting */
+	tm.second = seconds;
```

Comment:
`seconds` should be `uint64_t`. Please change `shell_strtol()` to `shell_strtoull()`